packages/next/src/client/components/react-dev-overlay/app/hot-reloader-client.tsx

```diff
@@ -554,6 +562,15 @@ export default function HotReload({
     }
   }, [dispatch])
 
+  //  We render a separate error overlay at the root when an error is thrown from rendering RSC, so
+  //  we should not render an additional error overlay in the descendent. However, we need to
+  //  keep rendering these hooks to ensure HMR works when the error is addressed.
+  const shouldRenderErrorOverlay = useSyncExternalStore(
+    () => () => {},
+    () => !shouldRenderRootLevelErrorOverlay(),
```

Comment:
I didn't fully get why `shouldRenderRootLevelErrorOverlay` isn't included in the client bundle, I feel like this is a bit of a workaround. We can follow up later!

Reply:
What do you mean by including in the client bundle?